### PR TITLE
ARP: implement decoder and logger v11

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)
- [x] I have created a ticket at
      https://redmine.openinfosecfoundation.org/projects/suricata/issues
      (if applicable)

Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/6827

Describe changes:
- Make checks stricter (check the entire fixed part of a packet)
- Add `PacketClearL3(p);`
- Remove opcodes that don't work with ethernet
- Set an event if an opcode is not supported
- Accept ARP packets longer than 28 bytes without setting an event.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1807

### CODE REVIEW

File: src/decode-arp.c
Comment (victorjulien) [inline src/decode-arp.c:77]: what do these mean?

File: src/decode-arp.c
Comment (victorjulien) [inline src/decode-arp.c:42]: I wonder if we could do something else here:
1. create a minimal header type + PACKET_L3_ARP_MIN or something
2. first set that, then update it with the full header after the checks below
3. update the logger to allow for logging only the minimal header if that is set

That way we add more value for unsupported types.

We also don't set an unchecked header pointer. It should be safe in your current code, as we unset it after checking, but it still feels like a safer pattern.

Can you do this as a follow up PR+ticket?

**Review by victorjulien — APPROVED** [review]

File: src/decode-arp.c
Comment (glongo) [reply src/decode-arp.c:77]: 1 = request
2 = reply
3 = inverse_request
4 = inverse_response

Do you want me to add some `#define` for these?

File: src/decode-arp.c
Comment (glongo) [reply src/decode-arp.c:42]: Yes, sure!

File: src/decode-arp.c
Comment (victorjulien) [reply src/decode-arp.c:77]: yeah or an enum. Since this PR is approved you can do it after merge in a follow up PR (unless the staging process flags big issues)

**Comment by victorjulien** [comment]
Merged in #11086, thanks!

File: src/decode-arp.c
Comment (jufajardini) [reply src/decode-arp.c:77]: Redmine ticket: https://redmine.openinfosecfoundation.org/issues/7030 :P

**Pull request closed without merging**
